src/kinsol/kinsol.c

```diff
@@ -3010,50 +2975,148 @@ static int KINFP(KINMem kin_mem)
  * ========================================================================
  */
 
+static int AndersonAccQRDelete(KINMem kin_mem, N_Vector* Q, sunrealtype* R,
+                               int depth)
+{
+  /* local dot product flag for single buffer reductions */
+  sunbooleantype dotprodSB = SUNFALSE;
+
+  if ((kin_mem->kin_vtemp2->ops->nvdotprodlocal ||
+       kin_mem->kin_vtemp2->ops->nvdotprodmultilocal) &&
+      kin_mem->kin_vtemp2->ops->nvdotprodmultiallreduce)
+  {
+    dotprodSB = SUNTRUE;
+  }
+
+  /* Delete left-most column vector from QR factorization */
+  sunrealtype a, b, temp, c, s;
+
+  for (int i = 0; i < depth - 1; i++)
+  {
+    a                          = R[(i + 1) * depth + i];
+    b                          = R[(i + 1) * depth + i + 1];
+    temp                       = SUNRsqrt(a * a + b * b);
```

Comment:
`hypot` would be more accurate. Looks like a Givens rotation, and there's a few tricks to improve the numerical stability: https://en.wikipedia.org/wiki/Givens_rotation#Stable_calculation. Not sure how important...

Reply:
Something to consider in the future (maybe when updating this to remove multiple columns). For now this function just encapsulates the exiting functionality.